src/app/authoring-tool/edit-dynamic-prompt/edit-dynamic-prompt.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { DynamicPrompt } from '../../../assets/wise5/directives/dynamic-prompt/DynamicPrompt';
+import { ProjectService } from '../../../assets/wise5/services/projectService';
+
+@Component({
+  selector: 'edit-dynamic-prompt',
+  templateUrl: './edit-dynamic-prompt.component.html',
+  styleUrls: ['./edit-dynamic-prompt.component.scss']
+})
+export class EditDynamicPromptComponent implements OnInit {
+  allowedConnectedComponentTypes: string[] = ['OpenResponse'];
+  @Input() authoringComponentContent: any;
+  @Output() dynamicPromptChangedEvent = new EventEmitter<string>();
+
+  constructor(private projectService: ProjectService) {}
+
+  ngOnInit(): void {}
+
+  toggleDynamicPrompt(event: any): void {
+    if (this.authoringComponentContent.dynamicPrompt == null) {
+      this.authoringComponentContent.dynamicPrompt = new DynamicPrompt({
+        referenceComponent: {},
+        rules: []
+      });
+    }
+    this.authoringComponentContent.dynamicPrompt.enabled = event.checked;
+    this.dynamicPromptChangedEvent.next(event);
```

Comment:
This component's parent component doesn't do anything with this emitted event right now, so we can just emit a void?
```suggestion
    this.dynamicPromptChangedEvent.next();
```